src/components/button/code/templates/buttonTemplate.jsx

```diff
@@ -2,14 +2,15 @@ var React = require('react');
 var classNames = require('classnames');
 var getComponentClasses = require('../../../../utils/getComponentClasses');
 
-module.exports = function(props) {
+module.exports = function(context) {
```

Comment:
the use of the variable name `context` might be confusing when/if we start using React's contexts. It's currently undocumented, but is available to use. This won't stop me +1ing this, but I thought it might be worth mentioning.

https://facebook.github.io/react/blog/2014/03/28/the-road-to-1.0.html#context
https://www.tildedave.com/2014/11/15/introduction-to-contexts-in-react-js.html

*As an alternative* we could in the view render do:
```
render: function() {
  if(this.props.href) {
    return require('../templates/buttonAnchorTemplate.jsx').call(this);
  }
  return require('../templates/buttonTemplate.jsx').call(this);
}
```

then the above function could be written as:
```
module.exports = function() {

  var propClasses = ['size', 'purpose'];
  var classes = getComponentClasses('component-button', propClasses, this.props);
  var dataAttributes = this.getDataAttributesFromProps();
  
  return (
    <button className={classNames(classes)} disabled={this.props.disabled} type={this.props.type} onClick={this.props.handleClick} {...dataAttributes}>
      {this.props.children}
    </button>
  );
};
```

(note the use of _this_ instead of a passed in variable)

What do people think of that as an alternative? (if it's preferred then we should probably put a JIRA in to change all the view->template relationships to work like this)